packages/@lwc/engine/src/framework/component.ts

```diff
@@ -56,6 +57,7 @@ export function getComponentRegisteredMeta(Ctor: ComponentConstructor): Componen
 
 export function createComponent(uninitializedVm: UninitializedVM, Ctor: ComponentConstructor) {
     // create the component instance
+    Ctor = getComponentOrSwappedComponent(Ctor);
```

Comment:
This should be protected with a `process.env !== 'production'` branch

Reply:
Good call, that was me yesterday fixing the problem. @caridy you want me to fix it or you want to do another pass?

Reply:
yeah, I'm changing the position were this is actually going to happen, will submit the PR in a bit

Reply:
resolved by moving to another place and adding the condition.